overlord/servicestate/service_control.go

```diff
@@ -54,6 +55,7 @@ type ServiceAction struct {
 	// inactive and not disabled services in snap-name, and also svc1 regardless
 	// of the state svc1 is in.
 	ExplicitServices []string `json:"explicit-services,omitempty"`
+	DisabledServices []string `json:"disabled-services,omitempty"`
```

Comment:
I'm not sure it's a good idea to query disabled services when creating the task in serviceControlTs and store them in ServiceAction on the task - the task may be executed some time later after it was created. I think we should query for disabled services when we handle the task, close to where they need to be passed to wrappers to minimize the time window between checking and acting on them.

Reply:
Fixed!